File: Stack/stack.js
const MAX = 4;
let stack_arr = [MAX]
let top = -1;

let  isEmpty = function() {
    if(top=== -1)
        return 1;
    else
        return 0;
}

let isFull = function() {
    if(top===MAX-1)
        return 1;
    else
        return 0;
}

let peek = function() {
    return stack_arr[top]
}

let stack_push = function(data) {
    if(isFull())
        console.log("stack overflow");
    top = top +1;
    stack_arr[top] = data;
}

let stack_pop = function() {
    if(isEmpty()) {
        console.log("stack is empty");
    } else {
    
    stack_arr[top];
    top = top-1;
    }
}

let print = function() {
    for(let i=top; i>=0; i--) {
        console.log(stack_arr[i]);
    }
}

stack_push(1)
stack_push(2)
stack_push(3)



print()

File: Stack/stack.c
#include<stdio.h>
#include<stdlib.h>
#include<limits.h>

struct Stack {
    int top;
    unsigned cap;
    int* arr;
};
typedef struct Stack st;

st* create_stack(unsigned cap)
{
    st* stack = malloc(sizeof(st));
    stack->cap = cap;
    stack->top = -1;
    stack->arr = malloc(sizeof(st));
    return stack;
}

int isFull(st* stack)
{
    return stack->top == stack->cap - 1;
}

int isEmpty(st* stack)
{
    return stack->top == -1;
}

void push(st* stack, int data)
{
    if(isFull(stack))
        return;
    stack->arr[++stack->top] == data;
    printf("%d pushed to stack\n", data);
}

int pop(st* stack)
{
    if(isEmpty(stack))
        return INT_MIN;
    return stack->arr[stack->top--];
}


int main(int argc, char const *argv[])
{
    st* stack = create_stack(3);
    push(stack, 10);
    push(stack, 20);
    push(stack, 30);
    push(stack, 30);

    printf("%d ",pop(stack));
    push(stack, 30);

    
 

    

    return 0;
}
